src/codegen/baseline_jit.cpp

```diff
@@ -870,8 +872,13 @@ Box* JitFragmentWriter::createListHelper(uint64_t num, Box** data) {
 
 Box* JitFragmentWriter::createSetHelper(uint64_t num, Box** data) {
     BoxedSet* set = (BoxedSet*)createSet();
-    for (int i = 0; i < num; ++i)
-        set->s.insert(data[i]);
+    for (int i = 0; i < num; ++i) {
+        auto&& p = set->s.insert(data[i]);
+        if (!p.second /* already exists */) {
+            Py_DECREF(p.first->value);
+            *p.first = data[i];
```

Comment:
This matches the behavior for dict keys, but is different from dict values -- `{1:1, 1L:1L}` is `{1: 1L}`.